yesod-test/test/main.hs

```diff
@@ -776,6 +814,48 @@ getIntegerR = do
     app <- getYesod
     pure $ T.pack $ show (routedAppInteger app)
 
+tupleField = Field parse view UrlEncoded
+  where
+  parse :: [Text] -> [FileInfo] -> (HandlerFor RoutedApp) (Either (SomeMessage (HandlerSite (HandlerFor RoutedApp))) (Maybe (Int, Text)))
+  parse rawVals@[someNumber, someText] _fileVals =  
+    case readMaybe (T.unpack someNumber)::Maybe Int of
+      Nothing -> pure $ Left "The first element is not Int"
+      Just n -> pure $ Right $ Just (n, someText)
+  parse rawVals _ = pure $ Left "The number of elements in the first argument of the parse is not equal to two"
+
+  view theId name attrs _eVal isReq =
+    [whamlet|
+      <input name=#{name}>
+      <input name=#{name}>
+    |]
+
+tupleForm :: Html -> MForm Handler (FormResult (Int, Text), Widget)
+tupleForm = renderDivs $ areq tupleField "" { fsName = Just "data" } Nothing
+
+getTupleR :: Handler Html
+getTupleR = do
+    ((result, widget), enctype) <- runFormPost tupleForm
+    defaultLayout
+        [whamlet|
+            <form method=post action=@{TupleR} enctype=#{enctype}>
+                ^{widget}
+                <button>Submit
+        |]
+
+postTupleR :: Handler Html
+postTupleR = do
+    ((result, widget), enctype) <- runFormPost tupleForm
+    case result of
+        FormSuccess tupleValue -> defaultLayout [whamlet|<p>#{show tupleValue}|]
+        FormFailure errors -> invalidArgs errors
+        FormMissing ->
+           defaultLayout
+             [whamlet|
+                <p>Form is missing, let's try again.
+                <form method=post action=@{TupleR} enctype=#{enctype}>
+                    ^{widget}
+                    <button>Submit
+             |]
```

Comment:
I don't think this is relevant to the test, so it may as well be something like this.

```suggestion
        FormMissing -> error "Tuple form is missing"
```